turbo/app/import.go

```diff
@@ -205,6 +207,18 @@ func missingBlocks(chainDB kv.RwDB, blocks []*types.Block) []*types.Block {
 }
 
 func InsertChain(ethereum *eth.Ethereum, chain *core.ChainPack) error {
+	tx, err := ethereum.ChainDB().BeginRo(context.Background())
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	// Check the validity of incoming chain - only in case of bor consensus
+	currentHead := rawdb.ReadCurrentHeader(tx)
+	if isValid, _ := borfinality.ValidateReorg(currentHead, chain.Headers, ethereum.ChainConfig()); !isValid {
+		return whitelist.ErrMismatch
```

Comment:
We should think on improving error handling here. Also, the errors populated in the child functions are not populated here well.